app/Main.hs

```diff
@@ -7,12 +7,16 @@ import Web.Spock.Config
 import Control.Monad.IO.Class (liftIO)
 import Data.Yaml.Config (loadYamlSettings, useEnv)
 import Data.Aeson (parseJSON, FromJSON, withObject, (.:))
+import Database.Persist.MySQL (MySQLConf)
 
-data AppSettings = AppSettings { appPort :: Int }
+data AppSettings = AppSettings { appPort :: Int
+                               , appDatabaseConf :: MySQLConf
+                               }
 
 instance FromJSON AppSettings where
   parseJSON = withObject "AppSettings" $ \o -> do
     appPort <- o .: "port"
+    appDatabaseConf <- o .: "database"
     return AppSettings{..}
```

Comment:
o is for Object (it is no longer json). Also the Haskell way is to use short names when types can tell you what you want to know. We can break that convention, but most examples are going to look like what I've done here for parsing JSON.